c_binding/include/libstoragemgmt/libstoragemgmt_nfsexport.h

```diff
@@ -73,53 +73,130 @@ lsm_nfs_export LSM_DLL_EXPORT **
 
 
 /**
- * Frees the memory for a NFS export record.
- * @param exp
- * @return Error code as enumerated by \ref lsm_error_number.
- * @retval LSM_ERR_OK on success.
+ * lsm_nfs_export_record_free - Free the lsm_nfs_export memory.
+ *
+ * Version:
+ *      1.0
+ *
+ * Description:
+ *      Frees the memory resources for the specified lsm_nfs_export.
+ *
+ * @exp:
+ *      The pointer of lsm_nfs_export to free.
+ *
+ * Return:
+ *      Error code as enumerated by 'lsm_error_number':
+ *          * LSM_ERR_OK
+ *              On success.
+ *          * LSM_ERR_INVALID_ARGUMENT
+ *              When not a valid lsm_disk pointer.
  */
 int LSM_DLL_EXPORT lsm_nfs_export_record_free(lsm_nfs_export *exp);
 
 /**
- * Frees the memory for the NFS export array and the memory for each entry
- * @param exps          Memory to free
- * @param size          Number of entries
- * @return Error code as enumerated by \ref lsm_error_number.
- * @retval LSM_ERR_OK on success.
- *  */
+ * lsm_nfs_export_record_array_free - Free the memory of lsm_nfs_export array.
+ *
+ * Version:
+ *      1.0
+ *
+ * Description:
+ *      Frees the memory resources for an array of lsm_nfs_export.
+ *
+ * @exps:
+ *      Array to release memory for.
+ * @size:
+ *      Number of elements.
+ * Return:
+ *      Error code as enumerated by 'lsm_error_number':
+ *          * LSM_ERR_OK
+ *              On success.
+ *          * LSM_ERR_INVALID_ARGUMENT
+ *              When not a valid lsm_nfs_export pointer.
+ *
+ */
 int LSM_DLL_EXPORT lsm_nfs_export_record_array_free(lsm_nfs_export * exps[],
                                                     uint32_t size);
 
 /**
- * Duplicates the source and returns the copy.
- * @param source            Source record to copy
- * @return Copy of source, else NULL one error.
+ * lsm_nfs_export_record_copy - Duplicates a lsm_nfs_export record.
+ *
+ * Version:
+ *      1.0
+ *
+ * Description:
+ *      Duplicates a lsm_nfs_export record.
+ *
+ * @source:
+ *      Pointer of lsm_nfs_export to duplicate.
+ *
+ * Return:
+ *      Pointer of lsm_nfs_export. NULL on memory allocation failure. Should be
+ *      freed by lsm_nfs_export_record_free().
  */
 lsm_nfs_export LSM_DLL_EXPORT *
     lsm_nfs_export_record_copy(lsm_nfs_export *source);
 
 /**
- * Returns the ID
- * @param exp       Valid nfs export record
- * @return Pointer to ID
+ * lsm_nfs_export_id_get - Retrieves the ID of the nfs_export.
+ *
+ * Version:
+ *      1.0
+ *
+ * Description:
+ *      Retrieves the ID for the nfs_export.
+ *      Note: Address returned is valid until lsm_nfs_export gets freed, copy
+ *      return value if you need longer scope. Do not free returned string.
+ *
+ * @exp:
+ *      NFS export to retrieve id for.
+ *
+ * Return:
+ *      string. NULL if argument 'exp' is NULL or not a valid lsm_nfs_export
+ *      pointer.
  */
 const char LSM_DLL_EXPORT *lsm_nfs_export_id_get(lsm_nfs_export *exp);
 int LSM_DLL_EXPORT lsm_nfs_export_id_set(lsm_nfs_export *exp,
```

Comment:
These functions are plugin only function, I don't want to expose the documents of those plugin only functions yet.

It should be in plugin interface header, but due to ABI concern, we just skip the user documentation for it.